### DISCUSSION TITLE
For production deployments, is it generally okay to have more than one dagster daemon instance?

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#17542] [Q&A] [answered]
Quick question: For production deployments, is it generally okay to have more than one dagster daemon instance? From what I understand, they're used to kickoff various background tasks like sensors and auto-materialization of assets. What safeguards would be in place to handle race conditions between multiple dagster daemon instances such as two of them picking up the same task?

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1698780432004019?thread_ts=1698780432.004019&cid=C01U954MEER).*

**Comment by garethbrickman** [comment, marked as answer]
No. "A deployment can have multiple simultaneous instances of dagster-webserver, but should include only a single dagster-daemon process." From [Running the Dagster daemon](https://docs.dagster.io/deployment/guides/service#running-the-dagster-daemon) documentation.

Having multiple instances of the `dagster-daemon` could lead to race conditions and other issues.
